backend/modules/database/db_functions.bal

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2025 WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License. 
+// import ballerina/sql;
+
+# Fetch all apps visible to the given `roles`
+#
+# + email - User email used to look up favourites
+# + roles - Role names used to resolve visible apps
+# + return - App[] with `isFavourite` set, or an `error?` on failure
+public isolated function fetchAppByRoles(string email, string[] roles) returns App[]|error {
```

Comment:
Database function name and query name are different. Shall we make one name?
`fetchAppByRoles` and `fetchAppByRolesQuery`